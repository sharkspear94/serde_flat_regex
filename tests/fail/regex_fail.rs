use serde::Deserialize;
use serde_flat_regex::{self, flat_regex};

#[flat_regex]
#[derive(Debug, Deserialize, PartialEq, Clone)]
struct Foo {
    id: u32,
    #[flat_regex(regex = r"[a-Z]")]
    rest: std::collections::HashMap<std::string::String, String>,
}

fn main() {}
