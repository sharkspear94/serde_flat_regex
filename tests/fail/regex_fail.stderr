error: regex parse error:
           [a-Z]
            ^^^
       error: invalid character class range, the start must be <= the end
 --> tests/fail/regex_fail.rs:8:17
  |
8 |     #[flat_regex(regex = r"[a-Z]")]
  |                 ^^^^^^^^^^^^^^^^^^
